pulp_rpm/tests/functional/api/test_sync.py

```diff
@@ -75,1390 +66,988 @@
-
-    @skip_if(bool, "md5_allowed", True)
-    def test_sync_metadata_with_unsupported_checksum_type(self):
-        """
-        Sync an RPM repository with an unsupported checksum (md5).
-
-        This test require disallowed 'MD5' checksum type from ALLOWED_CONTENT_CHECKSUMS settings.
-        """
-        with self.assertRaises(PulpTaskError) as ctx:
-            self.do_test(RPM_MD5_REPO_FIXTURE_URL)
-
-        self.assertIn(
-            "does not contain at least one trusted hasher which is specified in the "
-            "'ALLOWED_CONTENT_CHECKSUMS'",
-            ctx.exception.task.error["description"],
-        )
+    assert any(report.code == "sync.was_skipped" for report in task.progress_reports)
 
-    @unittest.skip(
-        "Needs a repo where an unacceptable checksum is used for packages, but not for metadata"
+    repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href, optimize=False)
+    sync_response = rpm_repository_api.sync(repository.pulp_href, repository_sync_data)
+    task = monitor_task(sync_response.task)
+
+    assert all(report.code != "sync.was_skipped" for report in task.progress_reports)
+
+    # create a new repo version, sync again, assert not optimized
+    repository = rpm_repository_api.read(repository.pulp_href)
+    content = choice(get_content(repository.to_dict())[RPM_PACKAGE_CONTENT_NAME])
+    response = rpm_repository_api.modify(
+        repository.pulp_href, {"remove_content_units": [content["pulp_href"]]}
     )
-    @skip_if(bool, "md5_allowed", True)
-    def test_sync_packages_with_unsupported_checksum_type(self):
-        """
-        Sync an RPM repository with an unsupported checksum (md5) used for packages.
-
-        This test require disallowed 'MD5' checksum type from ALLOWED_CONTENT_CHECKSUMS settings.
-        """
-        with self.assertRaises(PulpTaskError) as ctx:
-            self.do_test(RPM_MD5_REPO_FIXTURE_URL)
-
-        self.assertIn(
-            "rpm-with-md5/bear-4.1-1.noarch.rpm contains forbidden checksum type",
-            ctx.exception.task.error["description"],
-        )
+    monitor_task(response.task)
 
-    def test_complete_mirror_with_xml_base_fails(self):
-        """Test that syncing a repository that uses xml:base in mirror mode fails."""
-        with self.assertRaises(PulpTaskError) as ctx:
-            self.do_test(REPO_WITH_XML_BASE_URL, sync_policy="mirror_complete")
+    repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
+    sync_response = rpm_repository_api.sync(repository.pulp_href, repository_sync_data)
+    task = monitor_task(sync_response.task)
 
-        self.assertIn(
-            "features which are incompatible with 'mirror' sync",
-            ctx.exception.task.error["description"],
-        )
+    assert all(report.code != "sync.was_skipped" for report in task.progress_reports)
 
-    @unittest.skip("Repo URL no longer works")
-    def test_complete_mirror_with_external_location_href_fails(self):
```

Comment:
I'm hoping we can restore this with fixtures later, let's keep it please.  At minimum file an issue for its restoration

Reply:
It is there. I did not remove it.

Reply:
I just saw that the constant was missing

Reply:
The constant was removed because it is worthless at the moment, is not it?

Reply:
It's fine